webpack/JobInvocationDetail/TemplateInvocation.js

```diff
@@ -54,28 +59,39 @@ export const TemplateInvocation = ({
   isInTableView,
   hostName,
   hostProxy,
+  isExpanded,
 }) => {
   const intervalRef = useRef(null);
   const templateURL = showTemplateInvocationUrl(hostID, jobID);
   const hostDetailsPageUrl = useForemanHostDetailsPageUrl();
-  const { response, status, setAPIOptions } = useAPI('get', templateURL, {
-    key: GET_TEMPLATE_INVOCATION,
-    headers: { Accept: 'application/json' },
-    handleError: () => {
-      if (intervalRef.current) {
-        clearInterval(intervalRef.current);
-        intervalRef.current = null;
-      }
-    },
-  });
-  const { finished, auto_refresh: autoRefresh } = response;
+
+  const status = useSelector(selectTemplateInvocationStatus(hostID));
+  const response = useSelector(selectTemplateInvocation(hostID));
+  const finished = response.finished || true;
+  const autoRefresh = response.auto_refresh || false;
+  const dispatch = useDispatch();
 
   useEffect(() => {
+    const getData = async () => {
```

Comment:
Is it okay to define a function in `useEffect` hook?..

Reply:
it is recommended